test/functional/multiwallet.py

```diff
@@ -16,85 +16,101 @@ class MultiWalletTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [['-wallet=w1', '-wallet=w2', '-wallet=w3', '-wallet=w']]
 
     def run_test(self):
-        assert_equal(set(self.nodes[0].listwallets()), {"w1", "w2", "w3", "w"})
+        # helpers to construct data paths
+        data_dir = lambda *p: os.path.join(self.options.tmpdir, 'node0', 'regtest', *p)
+        wallet_dir = lambda *p: data_dir('wallets', *p)
 
+        # check wallet.dat is created
         self.stop_node(0)
+        assert_equal(os.path.isfile(wallet_dir('wallet.dat')), True)
+
+        # create symlink to verify wallet directory path can be referenced
+        # through symlink
+        os.mkdir(wallet_dir('w7'))
+        os.symlink('w7', wallet_dir('w7_symlink'))
+
+        # rename wallet.dat to make sure plain wallet file paths (as opposed to
+        # directory paths) can be loaded
+        os.rename(wallet_dir("wallet.dat"), wallet_dir("w8"))
+
+        # restart node with a mix of wallet names:
+        #   w1, w2, w3 - to verify new wallets created when non-existing paths specified
+        #   w          - to verify wallet name matching works when one wallet path is prefix of another
+        #   sub/w5     - to verify relative wallet path is created correctly
+        #   extern/w6  - to verify absolute wallet path is created correctly
+        #   w7_symlink - to verify symlinked wallet path is initialized correctly
+        #   w8         - to verify existing wallet file is loaded correctly
+        #   ''         - to verify default wallet file is created correctly
```

Comment:
I don't think there's any way to access the 'default' wallet with name `''` through the CLI.

```
(Pdb) self.nodes[0].cli("--rpcwallet=").getwalletinfo()
*** subprocess.CalledProcessError: Command 'bitcoin-cli' returned non-zero exit status 18
```